loadtest/control/simulcontroller/actions.go

```diff
@@ -28,18 +28,37 @@ func (c *SimulController) connect() error {
 	if err != nil {
 		return fmt.Errorf("connect failed %w", err)
 	}
+	c.connected <- struct{}{}
 	go func() {
 		for err := range errChan {
 			c.status <- c.newErrorStatus(err)
 		}
 	}()
+	go c.wsEventHandler()
+	go c.periodicActions()
+	return nil
+}
 
+func (c *SimulController) disconnect() error {
+	// one for ws loop and one for periodic actions loop
+	select {
+	case <-c.connected:
```

Comment:
Same as above, we should do a CAS and then move on. This is un-idiomatic and error-prone. What happens if one calls connect consecutively twice ? Or even calls disconnect twice ? :)

Either we use CAS, or not. 0/5 on that. But we shouldn't use buffered channels to track state.